### PR TITLE
py-shop-trip-v0

### PR DESCRIPTION
**Pull request by muriakula**
I have a problem with pytest. pytest cannot find the config.json file

### CODE REVIEW

File: app/car.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+@dataclass
+class Car:
+    brand: str
+    fuel_consumption: float
+
+    @classmethod
+    def set_customers_car(cls, car: dict) -> "Car":
+        return cls(car["brand"], car["fuel_consumption"])
```
Comment (olena-hul) [inline app/car.py:11]: rename method to `create`

File: app/customer.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@dataclass
+class Customer:
+    name: str
+    product_cart: dict[str: int]
+    location: list[int]
+    money: int
+    car: Car
+
+    @classmethod
+    def set_customer_from_file(cls, customers: dict) -> list:
```
Comment (olena-hul) [inline app/customer.py:14]: Use better method name

File: app/shop.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list[int]
+    products: dict[str: int | float]
+
+    @classmethod
+    def set_shop_from_file(cls, shops: dict) -> list:
```
Comment (olena-hul) [inline app/shop.py:11]: Use better method name

File: app/json_data.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+def distance(point1: list, point2: list) -> float:
+    return math.sqrt(
+        (point2[0] - point1[0]) ** 2
+        + (point2[1] - point1[1]) ** 2
+    )
+
+
+def shopping(customer: Customer, shop: Shop) -> int:
+    ttl_product_price = 0
```
Comment (olena-hul) [inline app/json_data.py:25]: Use full names, ttl is commonly used to specify `time to live`, not total

**Review by olena-hul — CHANGES_REQUESTED** [review]

File: app/customer.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+@dataclass
+class Customer:
+    name: str
+    product_cart: dict[str: int]
+    location: list[int]
+    money: int
+    car: Car
+
+    @classmethod
+    def create_customers(cls, customers: dict) -> list:
```
Comment (Bohdan-Salamakha) [inline app/customer.py:14]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]

File: app/shop.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+
+@dataclass
+class Shop:
+    name: str
+    location: list[int]
+    products: dict[str: int | float]
+
+    @classmethod
+    def create_shops(cls, shops: dict) -> list:
```
Comment (Bohdan-Salamakha) [inline app/shop.py:11]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,39 @@
...
+            the_best_shop = distance_cost[min_shop_cost]
+            print(f"{customer.name} rides to {the_best_shop.name}\n"
+                  f"\nDate: "
+                  f"{datetime.datetime.now().strftime('%d/%m/%Y %H:%M:%S')}\n"
+                  f"Thanks, {customer.name}, for your purchase!\n"
+                  f"{the_best_shop.receipt}\nSee you again!\n"
+                  f"\n{customer.name} rides home\n"
+                  f"{customer.name} now has "
+                  f"{customer.money - min_shop_cost} dollars\n"
+                  )
```
Comment (Bohdan-Salamakha) [inline app/main.py:39]:
```suggestion
            print(
                  f"{customer.name} rides to {the_best_shop.name}\n"
                  f"\nDate: "
                  f"{datetime.datetime.now().strftime('%d/%m/%Y %H:%M:%S')}\n"
                  f"Thanks, {customer.name}, for your purchase!\n"
                  f"{the_best_shop.receipt}\nSee you again!\n"
                  f"\n{customer.name} rides home\n"
                  f"{customer.name} now has "
                  f"{customer.money - min_shop_cost} dollars\n"
                  )
```

File: app/json_data.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+def create_elements() -> tuple:
+    with open("app/config.json") as file:
+        data = json.load(file)
+    fuel_price = data["FUEL_PRICE"]
+    customers = Customer.create_customers(data["customers"])
+    shops = Shop.create_shops(data["shops"])
+    return fuel_price, customers, shops
+
+
+def distance(point1: list, point2: list) -> float:
```
Comment (Bohdan-Salamakha) [inline app/json_data.py:17]: add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contains string elements has typehint list[str]

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

**Review by Bohdan-Salamakha — APPROVED** [review]

**Pull request closed without merging**